File: bindings/java-jni/src/main/java/org/maplibre/nativejni/resource/ResourceStoragePolicy.java
package org.maplibre.nativejni.resource;

/** Resource storage policy copied from a native resource request. */
public enum ResourceStoragePolicy {
  PERMANENT(0),
  VOLATILE(1),
  UNKNOWN(-1);

  private final int nativeValue;

  ResourceStoragePolicy(int nativeValue) {
    this.nativeValue = nativeValue;
  }

  public int nativeValue() {
    return nativeValue;
  }

  public static ResourceStoragePolicy fromNative(int nativeValue) {
    return switch (nativeValue) {
      case 0 -> PERMANENT;
      case 1 -> VOLATILE;
      default -> UNKNOWN;
    };
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/resource/ResourceStoragePolicy.kt
package org.maplibre.nativeffi.resource

/** Resource storage policy copied from a native resource request. */
public enum class ResourceStoragePolicy(public val nativeValue: Int) {
  PERMANENT(0),
  VOLATILE(1),
  UNKNOWN(-1);

  public companion object {
    internal fun fromNative(nativeValue: UInt): ResourceStoragePolicy =
      fromNative(nativeValue.toInt())

    public fun fromNative(nativeValue: Int): ResourceStoragePolicy =
      entries.firstOrNull { it.nativeValue == nativeValue } ?: UNKNOWN
  }
}
